### PR TITLE
Add description of consensus phases to README

### LINKED ISSUE #38: Update README with description of consensus phases
Add a short explanation of how SA works, describing the different phases.

### PR DESCRIPTION
**Pull request by fed-franz**
fixes dusk-network/consensus#38

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,19 +4,28 @@ The Dusk Network utilizes a consensus protocol called Succinct Attestation (SA).
 
 SA is permissionless, meaning that any eligible participant in the Dusk Network protocol can join and participate in the consensus process. To be eligible, a participant must meet the following requirements:
 
- - Have a pre-configured amount of DUSK locked as a stake (referred to as a `Provisioner`)
- - Have a stake with a maturity of at least two epochs (referred to as an `Eligible Provisioner`)
+ - Have a pre-configured amount of DUSK locked as a stake (referred to as a _Provisioner_)
+ - Have a stake with a maturity of at least two epochs (referred to as an _Eligible Provisioner_)
 
-Other terms used in the context of SA:
+The SA consensus is divided in _rounds_, each of which creates a new block. In turn, in each round, on or more _iterations_ of the following _phases_ are executed:
```
Comment (herr-seppia) [inline README.md:10]:
```suggestion
The SA consensus is divided in _rounds_, each of which creates a new block. In turn, in each round, one or more _iterations_ of the following _phases_ are executed:
```

File: README.md [hunk 2]
```diff
@@ -133,5 +142,6 @@ USAGE:
...
-[^4]: In the Selection phase, a replica (or Provisioner in Dusk terminolody) acting as a Block Generator, proposes a block. In the First Reduction phase, a sampled committee of Provisioners vote on the validity of the block by signing a message with a BLS signature and broadcasting it. In the Second Reduction phase, a different sampled committee of Provisioners vote on the block, gather the results of the First Reduction phase, and produce an "Agreement" message. The steps keep iterating until the process is interrupted by the `Agreement` loop. This allows the group to reach a final decision about which block to agree on.
+[^3]: <!-- TODO: add short description here --> Deterministic Sortition is described [here](https://wiki.dusk.network/en/deterministic-sortition).
```
Comment (herr-seppia) [inline README.md:145]: Leaving a comment here, to remind us to open a followup issue

File: README.md [hunk 3]
```diff
@@ -133,5 +142,6 @@ USAGE:
...
+[^5]: In the Selection phase, a replica (or Provisioner in Dusk terminolody) acting as a Block Generator, proposes a block. In the First Reduction phase, a sampled committee of Provisioners vote on the validity of the block by signing a message with a BLS signature and broadcasting it. In the Second Reduction phase, a different sampled committee of Provisioners vote on the block, gather the results of the First Reduction phase, and produce an "Agreement" message. The steps keep iterating until the process is interrupted by the `Agreement` loop. This allows the group to reach a final decision about which block to agree on.
```
Comment (herr-seppia) [inline README.md:147]: The message signed is the Hash of the block, maybe it worth to mention it

Comment (fed-franz) [reply README.md:147]: Actually, should we remove this [^5]  (or integrate it in the description)? It basically explains the same thing.
Then we can add a new note explaining the message thing, or mention it in the text

File: README.md (hunk 2 above)
Comment (fed-franz) [reply README.md:145]: Opened Issue #41 to address this.

File: README.md (hunk 3 above)
Comment (fed-franz) [reply README.md]: fixed in [ab09e71](https://github.com/dusk-network/consensus/pull/39/commits/ab09e718642de61e39dccaf908391ab019457a50)

File: README.md (hunk 1 above)
Comment (fed-franz) [reply README.md]: fixed (https://github.com/dusk-network/consensus/pull/39/commits/684e7e6f39495aa0668390c0708e9124a2e9a675)

**Review by herr-seppia — APPROVED** [review]
LGTM

File: README.md (hunk 2 above)
Comment (herr-seppia) [reply README.md:145]: _N.B.: issue has been moved to dusk-network/rusk#904_

**Pull request merged**